packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -155,16 +170,20 @@ export class CheckboxBase extends React.Component<ICheckboxProps, ICheckboxState
 
   private _onChange = (ev: React.FormEvent<HTMLElement>): void => {
     const { disabled, onChange } = this.props;
-    const { isChecked } = this.state;
+    const { isChecked, isIndeterminate } = this.state;
 
     if (!disabled) {
-      if (onChange) {
-        onChange(ev, !isChecked);
-      }
-
-      if (this.props.checked === undefined) {
-        this.setState({ isChecked: !isChecked });
+      if (isIndeterminate) {
+        if (onChange) {
+          onChange(ev, !isChecked);
+        }
+        this.setState({ isIndeterminate: false });
+      } else {
+        if (onChange) {
+          onChange(ev, !isChecked);
+        }
```

Comment:
looks like line 182 - 184 are the same in both branches of the if clause .. could you simplify it to something like
```
if(onChange){ 
  onChange(ev, !isChecked)
}

if(isIndeterminate){
  this.setState({ isIndeterminate: false });
}
```